src/rotten-tomatoes/index.js

```diff
@@ -1,3 +1,61 @@
-function rottenTomatoes(grid) {}
+function rottenTomatoes(grid) {
+  let nRow = grid.length;
+  let nCol = grid[0].length;
+  let noFresh = 0;
+  let queue = [];
+  //Add the position of all rotten tomatoe and count the no of fresh tomatoes
+  for (let i = 0; i < nRow; i++) {
+    for (let j = 0; j < nCol; j++) {
+      if (grid[i][j] === 1) {
+        noFresh += 1;
+      } else if (grid[i][j] === 2) {
+        queue.push([i, j]);
+      }
+    }
+  }
+  let minutes = 0;
+  let fourDirection = [
+    [1, 0],
+    [-1, 0],
+    [0, 1],
+    [0, -1],
+  ]; //up,down,left,right
+  //loop as long as our queue length is not equal to  0 and no of fresh tomatoe greater than 0
+  while (queue.length>0 && noFresh>0) {
+    let size = queue.length;
+    /*check if the adjacent node to the rotten tomatoe is fresh. if yes turn into rotten then return the position
+   back to the queue
+   1. no of fresh is reduced
+   2. no of mintues is added
+   */
+
+    while (size--) {
```

Comment:
I think you can make do without this loop. The outer loop covers the same scenarios.